src/dcrypto/v1/v1_test.go

```diff
@@ -92,3 +94,37 @@ func TestRoundTrip(t *testing.T) {
 		}
 	}
 }
+
+func TestHash(t *testing.T) {
+	sizes := []int{24, 1024, 15872, 16364, 16384, 16394, 16896, 66560}
+	for _, size := range sizes {
+		h := md5.New()
+		t.Logf("Testing file of size: %db, with password: %s", size, password)
+		b, err := randBytes(size)
+		if err != nil {
+			t.Errorf("randBytes(%d) => %q; want nil", size, err)
+			continue
+		}
+		encReader, err := newEncryptReader(bytes.NewBuffer(b), password, salt, 1024)
+		if err != nil {
+			t.Errorf("NewEncryptReader() => %q; want nil", err)
+			continue
+		}
+		cipher, err := ioutil.ReadAll(io.TeeReader(encReader, h))
+		if err != nil {
+			t.Errorf("ioutil.ReadAll(*EncryptReader) => %q; want nil", err)
+			continue
+		}
+		want := h.Sum(nil)
+		h.Reset()
+		got, err := Hash(bytes.NewBuffer(b), bytes.NewBuffer(cipher[0:HeaderSize]), password, h)
```

Comment:
bytes.NewReader since we are only expecting reads and not doing any work to io.Writer-s.